### PR TITLE
Added test file with initial tests.

### PR DESCRIPTION
**Pull request by willjgriff**
Added test file with tests for `initialize()` and `addVaultToken()`, still needs tests for `removeVaultToken()` and `redeem()`. It includes a helpers file and DaoDeployment file to extract some behaviour. 

Added babel to enable JS ES6 in test files.

Added truffle compiler config to compile to Solidity v0.4.24.

Fixed a couple bugs preventing compilation.

Added openzeppelin-solidity library for ERC20 contract. (Don't forget to `npm i`).

I'm using truffle v5.0.14 not that it should matter but sometimes it does. You should be able to start `ganache-cli` then run tests in a separate terminal with `truffle test`.

### CODE REVIEW

**Comment by 0xGabi** [comment]
@fabriziovigevani and me will continue tomorrow with tests left. Thanks for set it up for us 🙏🏻

**Comment by 0xGabi** [comment]
Hey @willjgriff can you take a look to the progress we made over the weekend. Thanks

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+   "configurations": [
+      {
+         "type": "node",
+         "request": "launch",
+         "name": "truffle test debuggable",
+         "program": "${workspaceFolder}/node_modules/truffle/build/cli.bundled.js",
+         "args": ["test"]
+      }
+   ]
+}
```
Comment (willjgriff) [inline .vscode/launch.json]: Is this .vscode file necessary? Can it be added to .gitignore?

File: contracts/Redemptions.sol [hunk 2]
```diff
@@ -38,51 +39,57 @@ contract Redemptions is AragonApp {
...
 
         for (uint256 i = 0; i < vaultTokens.length; i++) {
-            redemptionAmount = _amount.mul(vault.balance(vaultTokens[i])).div(redeemableToken.totalSupply);
+            redemptionAmount = _amount.mul(vault.balance(vaultTokens[i])).div(redeemableToken.totalSupply());
             vault.transfer(vaultTokens[i], msg.sender, redemptionAmount);
         }
-
-        require(redeemableToken.destroyTokens(msg.sender, _amount), ERROR_CANNOT_DESTROY_TOKENS);
+        // minime.destroyTokens() never returns false, only reverts on failure
+        redeemableToken.destroyTokens(msg.sender, _amount);
```
Comment (willjgriff) [inline contracts/Redemptions.sol]: Nice, we can remove the 'ERROR_CANNOT_DESTROY_TOKENS' error.

File: test/RedemptionsTest.js [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+            it('reverts if removing token not present', async () => {
+                await assertRevert(redemptions.removeVaultToken(accounts[0]),'REDEMPTIONS_NOT_VAULT_TOKEN')
+            })
+        })
+
+        context('redeem(uint256 _amount)', () => {
+
+            const redemptionAmount = 20000
+            const vaultToken0Ammount = 45231
+            const vaultToken1Ammount = 20001
```
Comment (willjgriff) [inline test/RedemptionsTest.js]: Typo, vaultToken1Amount

File: test/RedemptionsTest.js [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+            const redeemer = accounts[0]
+        
+            beforeEach( async () => {
+                //transfer tokens to vault
+                await token0.transfer(vault.address,vaultToken0Ammount)
+                await token1.transfer(vault.address,vaultToken1Ammount)
+
+                //mint redeemableTokens to first two accounts
+                await redeemableToken.generateTokens(redeemer,redemptionAmount)
+                await redeemableToken.generateTokens(rootAccount,80000)
```
Comment (willjgriff) [inline test/RedemptionsTest.js]: Constant for the 80000?

File: test/RedemptionsTest.js [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+                assert.equal(token1Balance.toNumber(), expectedRedeemAmountToken1)
+            })
+
+            it('reverts if amount to redeem is zero', async () => {
+                await assertRevert(redemptions.redeem(0, { from:redeemer }),'REDEMPTIONS_CANNOT_REDEEM_ZERO')
+            })
+
+            it('reverts if amount to redeem exceeds account\'s balance', async () => {
+                await assertRevert(redemptions.redeem(redemptionAmount + 1, { from:redeemer }),'REDEMPTIONS_INSUFFICIENT_BALANCE')
+            })
```
Comment (willjgriff) [inline test/RedemptionsTest.js]: Could test the other `require` statements but it's up to you.

**Comment by 0xGabi** [comment]
As @fabriziovigevani  mention on his PR:

>We thought it will be best to have a reference to the token manager instance of the redemption token instead of having a direct reference. 
>There's still two tests missing from the Redeem functionality that im not sure how to test.

require(msg.sender != address(vault),);
require(msg.sender != address(tokenManager),);

**Pull request merged**